core/src/main/java/org/mule/config/ChainedThreadingProfile.java

```diff
@@ -83,7 +83,7 @@ public ChainedThreadingProfile(ThreadingProfile delegate)
      */
     public ChainedThreadingProfile(ThreadingProfile delegate, boolean dynamic)
     {
-        if (!dynamic)
+        if (!(dynamic || delegate instanceof DirectThreadingProfile))
```

Comment:
Can not "delegate instanceof DirectThreadingProfile"  be replaced with "!delegate.isDoThreading()"?